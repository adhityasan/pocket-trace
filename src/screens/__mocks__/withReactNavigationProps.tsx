import React from "react";

export const createTestScreenProps = (props?: object) => ({
  navigation: {
    state: { params: {} },
    dispatch: jest.fn(),
    goBack: jest.fn(),
    dismiss: jest.fn(),
    navigate: jest.fn(),
    openDrawer: jest.fn(),
    closeDrawer: jest.fn(),
    toggleDrawer: jest.fn(),
    getParam: jest.fn(),
    setParams: jest.fn(),
    addListener: jest.fn(),
    push: jest.fn(),
    replace: jest.fn(),
    pop: jest.fn(),
    popToTop: jest.fn(),
    isFocused: jest.fn()
  },
  ...props
});

const withReactNavigationProps = (Component: any) => {
  const navigationProps = createTestScreenProps();

  return <Component {...navigationProps} />;
};

export default withReactNavigationProps;
