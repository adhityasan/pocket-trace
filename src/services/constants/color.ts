export const COLOR_LIGHT = {
  LIGHT: "#fafafa",
  DARK: "#1c1e21",
  PRIMARY: "#5145a0",
  LIGHT_PRIMARY: "#6256b1",
  SECONDARY: "#d4cef7",
  LIGHT_SECONDARY: "#e7e3fb"
};
