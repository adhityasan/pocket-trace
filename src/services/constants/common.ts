export const WATCH_POSITION_TIME_INTERVAL = 1000; // milisecond
export const WATCH_POSITION_DISTANCE_INTERVAL = 10; // meter
