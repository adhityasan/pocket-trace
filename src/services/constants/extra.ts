import Constants from "expo-constants";

const EXTRA = Constants.manifest?.extra || {};

export const { APP_NAME = "$APP_NAME", APP_ENV } = EXTRA;

export default EXTRA;
