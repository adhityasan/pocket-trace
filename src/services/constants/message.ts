export const ERROR_MESSAGE = {
  signup_failed: "Oops, we have problem while signing you up",
  signin_failed: "Oops, we have problem while signing you in"
};
