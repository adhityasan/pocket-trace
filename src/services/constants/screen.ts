export enum SCREEN_NAMES {
  HOME = "Home",
  ABOUT = "About",
  AUTH_SCENE = "Auth Scene",
  RESOLVE_AUTH = "Resolve Auth",
  SIGNIN = "Signin",
  SIGNUP = "Signup"
}
