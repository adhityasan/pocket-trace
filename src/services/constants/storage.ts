export const ASYNC_STORAGE_KEYS = {
  TOKEN: "pocket_trace_token"
};
