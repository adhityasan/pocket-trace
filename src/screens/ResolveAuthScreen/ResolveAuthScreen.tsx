import React, { useEffect } from "react";
import { SafeAreaView, StyleSheet } from "react-native";

import { AuthStackProps } from "@@types/navigations/auth";
import AppIcon from "@components/AppIcon";
import { HeadingText } from "@components/Texts";
import useAuth from "@hooks/useAuth";
import { COLOR_LIGHT } from "@services/constants/color";
import { SCREEN_NAMES } from "@services/constants/screen";
import globalStyles from "@styles/_global";

const ResolveAuthScreen = ({ navigation }: AuthStackProps<SCREEN_NAMES.RESOLVE_AUTH>) => {
  const { resolveAuth } = useAuth();

  useEffect(() => {
    const resolve = async () => {
      const isAuthenticated = await resolveAuth();
      if (!isAuthenticated) {
        navigation.reset({ index: 0, routes: [{ name: SCREEN_NAMES.SIGNIN }] });
      }
    };
    resolve();
  }, []);

  return (
    <SafeAreaView style={[globalStyles.screenView, styles.screenView]}>
      <AppIcon color={COLOR_LIGHT.SECONDARY} />
      <HeadingText type="h1" style={styles.appNameText}>
        Pocket Trace
      </HeadingText>
    </SafeAreaView>
  );
};

const styles = StyleSheet.create({
  screenView: {
    backgroundColor: COLOR_LIGHT.DARK,
    alignItems: "center",
    justifyContent: "center"
  },
  appNameText: {
    color: COLOR_LIGHT.SECONDARY,
    marginTop: 10
  }
});

export default ResolveAuthScreen;
