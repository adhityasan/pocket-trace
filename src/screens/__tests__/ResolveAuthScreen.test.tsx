import AsyncStorage from "@react-native-async-storage/async-storage";
import { create, act } from "react-test-renderer";

import ResolveAuthScreen from "@screens/ResolveAuthScreen";
import withReactNavigationProps from "@screens/__mocks__/withReactNavigationProps";
import { ASYNC_STORAGE_KEYS } from "@services/constants/storage";

describe("screens/ResolveAuthScreen", () => {
  it("should match snapshot", () => {
    const withNavigationProps = withReactNavigationProps(ResolveAuthScreen);
    const result = create(withNavigationProps).toJSON();
    act(() => {
      expect(result).toMatchSnapshot();
    });
  });
  it("should get token from async storage", () => {
    const withNavigationProps = withReactNavigationProps(ResolveAuthScreen);
    act(() => {
      create(withNavigationProps);
      expect(AsyncStorage.getItem).toBeCalledWith(ASYNC_STORAGE_KEYS.TOKEN);
    });
  });
});
