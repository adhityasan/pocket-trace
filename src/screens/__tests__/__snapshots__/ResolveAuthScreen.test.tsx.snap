// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`screens/ResolveAuthScreen should match snapshot 1`] = `
<RCTSafeAreaView
  emulateUnlessSupported={true}
  style={
    Array [
      Object {
        "backgroundColor": "#fafafa",
        "flex": 1,
        "position": "relative",
      },
      Object {
        "alignItems": "center",
        "backgroundColor": "#1c1e21",
        "justifyContent": "center",
      },
    ]
  }
>
  <RNCMaskedView>
    <View
      pointerEvents="none"
      style={
        Object {
          "bottom": 0,
          "left": 0,
          "position": "absolute",
          "right": 0,
          "top": 0,
        }
      }
    >
      <Image
        source={1}
        style={
          Array [
            Object {
              "height": 50,
              "width": 50,
            },
            Object {},
          ]
        }
      />
    </View>
    <View
      style={
        Array [
          Object {
            "backgroundColor": "#d4cef7",
            "height": 50,
            "width": 50,
          },
          undefined,
        ]
      }
    />
  </RNCMaskedView>
  <Text
    style={
      Object {
        "color": "#d4cef7",
        "fontSize": 25,
        "fontWeight": "bold",
        "marginBottom": 3,
        "marginTop": 10,
      }
    }
  >
    Pocket Trace
  </Text>
</RCTSafeAreaView>
`;
